pkg/virtual/apiexport/builder/build.go

```diff
@@ -135,9 +138,36 @@ func BuildVirtualWorkspace(
 				kcpClusterClient,
 				wildcardKcpInformers.Apis().V1alpha1().APIResourceSchemas(),
 				wildcardKcpInformers.Apis().V1alpha1().APIExports(),
-				func(apiResourceSchema *apisv1alpha1.APIResourceSchema, version string, apiExportIdentityHash string) (apidefinition.APIDefinition, error) {
+				func(apiResourceSchema *apisv1alpha1.APIResourceSchema, version string, apiExport *apisv1alpha1.APIExport) (apidefinition.APIDefinition, error) {
+
+					if apiExport == nil {
+						return nil, fmt.Errorf("unable to get apiexport")
+					}
```

Comment:
testing every point for nil is an anti-pattern. This uses to be an invariant of a function signature.

Reply:
updated